### ISSUE TITLE
[FEATURE] Book club features

### ISSUE THREAD

**Issue by aaronleopold** [stumpapp/stump#120]
I think it would be really cute to add a suite of features aimed at facilitating book clubs

**Describe the solution you'd like**

- [x] User can create clubs
    - [x] Basic on/off RBAC, where a server admin may turn off this feature for a user. I think default in this case might be off.
- [ ] A user may invite another user to a club, where they are added **upon acceptance**
    -  NOTE: this requires a whole notification system 😬 
- [x] A club (entity) has:
    - [x] name + optional description
    - [x] members
    - [x] a book schedule
        - [x] configurable (but reasonably defaulted) interval between books (e.g. 1-2 month per book)
        - [x] book list with start/end dates for each (default interval, allow overriding)
- [ ] Clubs have a homepage
    - [ ] Current book (and next book) featured
    - [ ] User progress is displayed for each member of the club
        - [ ] Allow this to be turned off if a user would prefer others don't know progress
    - [ ] Basic chat board for each book that gets archived after each book rotation in schedule (**maybe**)
    - [ ] Admins of club have option to create/update book schedule

**Comment by DieselTech** [comment]
When the book club session is closed or ended, the users profile could show have many clubs they have participated in. I guess that also brings up a whole new feature centered around profile cards. But would be great in a social reading context. I feel like if there is a book club option that it would start to have a lot of users added to servers just for that.

![image](https://github.com/stumpapp/stump/assets/30128380/6db9472d-3832-441e-a39f-d3703c3689b3)

**Comment by aaronleopold** [comment]
> I guess that also brings up a whole new feature centered around profile cards. But would be great in a social reading context.

Yeah a whole new set of "public" user endpoints would have to be created to support this sort of thing. I like it though! Definitely something to keep in mind when I'm (finally, hopefully soon) able to give this feature some time. Once I finish implementing RBAC/ABAC Stump will be in a much better place for this kind of thing. One of my favorites though so super excited
